File: 1700/solution.ts
function countStudents(students: number[], sandwiches: number[]): number {
  const counts = [0, 0];
  for (const student of students) {
    counts[1] += student;
  }
  counts[0] = students.length - counts[1];
  for (let i = 0; i < sandwiches.length; i++) {
    if (counts[sandwiches[i]] == 0) {
      return sandwiches.length - i;
    }
    counts[sandwiches[i]]--;
  }
  return 0;
}

File: 1700/solution.go
package main

func countStudents(students []int, sandwiches []int) int {
	counts := [2]int{0, 0}
	for _, num := range students {
		counts[num]++
	}

	for i, num := range sandwiches {
		if counts[num] == 0 {
			return len(sandwiches) - i
		}
		counts[num]--
	}
	return 0
}